### ISSUE TITLE
How to insert records that are associated.

### ISSUE THREAD

**Issue by bas080** [2rhop/vollk#2]
Most of the time people use relational databases because they want to define a relationship between records by using some kind of identifier. Quite often developers choose to use the auto increment id to provide records with an id.

How would/could Vollk handle this usecase. Maybe nested specs?

```js
const user = {
  name: 'lorem.word',
  age: 'integer'
}

const profileSpec = {
  name: 'lorem.word',
  description: 'lorem.words',
  user: vollk.oneToOne('user', user),
  friends: vollk.oneToMany('user', user)
}
```

Under the hood this would generate multiple insertion per "item".

Also remember that there might be unique constraints. How to prevent randomly generated data from conflicting?

**Comment by 2rhop** [comment]
you are totally right! i guess there was more features to working on than i thought, i will appreciate any help to make vollk stronger

**Comment by 2rhop** [comment]
i want to separate vollk's code into 2 modules. The seeding part and cli part.
I want to name the seeding part **KNEX-SEEDER** to be more catchy and the cli just vollk but it will use the knex-seeder module

**Comment by bas080** [comment]
Knex has a schema builder. It would be nice if those definitions can be reused. These do not cover the associations use-case though. 

> example of such a schema

```js
knex.schema.createTable('users', function (table) {
  table.increments();
  table.string('name');
  table.integer('location_id');
  table.timestamps();
})
```

The simplest way to reuse these definitions without depending on the internal state of knex' schema instance internals would be to assign the table definition function to a variable. This function could then be reused in Vollk's api.

```js
volk.schema.seedTable('users', function(table) {...})
```

This api could optionally take a third argument that defines the association between tables, and it's schema. I am not sure how to do this correctly. I have something like this in mind.

```js
volk.schema.seedTable('users', function(table) {...}, function seedAssoc(user, assoc) {
  assoc.manyToOne('user_location', {
    user: user.location,
    location: assoc.oneToOne('location', createNewRecordRatio = 0.2)
  })
})
```

I think it's valuable to reuse code people might already have. It would be nice if Vollk's API can be used as a separate thing. As a result the code users already have require minimum change.

The assoc api proposal is a longshot. It requires way more scrutiny.

**Comment by 2rhop** [comment]
> in this very moment i am working exactly on that! thanks again for the **remarkable ideas!**

## soon i'll release the first version of [**`knex-seeder`**](https://github.com/2rhop/knex-seeder) (this is something like **vollk**'s api)

**Comment by 2rhop** [comment]
here is the first version https://github.com/2rhop/knex-seeder
